src/structures/Message.js

```diff
@@ -541,18 +541,27 @@ class Message extends Base {
 
   /**
    * Edits the content of the message.
-   * @param {?(string|APIMessage)} [content] The new content for the message
-   * @param {MessageEditOptions|MessageEmbed|MessageAttachment|MessageAttachment[]} [options] The options to provide
+   * @param {string|APIMessage|MessageEditOptions|MessageEmbed
+   * |MessageAttachment|MessageAttachment[]} options The options to provide
    * @returns {Promise<Message>}
    * @example
    * // Update the content of a message
    * message.edit('This is my new content!')
    *   .then(msg => console.log(`Updated the content of a message to ${msg.content}`))
    *   .catch(console.error);
    */
-  edit(content, options) {
-    options = content instanceof APIMessage ? content : APIMessage.create(this, content, options);
-    return this.channel.messages.edit(this.id, options);
+  edit(options) {
+    let data;
+
+    if (options instanceof APIMessage) {
+      data = options.resolveData();
+    } else if (typeof options === 'string') {
```

Comment:
This check isn't necessary with the current APIMessage, it's done in APIMessage.transformOptions (in reverse) currently. I think I may have confused you when I said you should check for strings. What I was trying to say is that transformOptions should perform this check and only ever be passed options and not content, options

Reply:
Thank you :), adressed